docs/integrations/destinations/s3-migrations.md

```diff
@@ -0,0 +1,202 @@
+| `{"type": "string", "format": "date" }`                                                    | `Optional Int32(logicalType=Date)`                                     | `date`                       |
+| `{"type": "string", "format": "time", "airbyte_type": "time_with_timezone" }`              | `Optional Int64(Optional Int64(logical_type=Time(Microseconds))`       | `time_with_timezone`         |
+| `{"type": "string", "format": "time", "airbyte_type": "time_without_timezone" }`           | `Optional Int64(Optional Int64(logical_type=Time(Microseconds))`       | `time_without_timezone`      |
+| `{"type": "string", "format": "date-time", "airbyte_type": "timestamp_with_timezone" }`    | `Optional Int64(logical_type=Timestamp(Microseconds)))`                | `timestamp_with_timezone`    |
+| `{"type": "string", "format": "date-time", "airbyte_type": "timestamp_without_timezone" }` | `Optional Int64(Optional Int64(logical_type=Timestamp(Microseconds)))` | `timestamp_without_timezone` |
+
+#### Merging like options in unions
+
+The above behavior applies only to unions of distinct types. Unions of the same type will continue to be merged. If the result is a single option, the union will be demoted to a field of that type. For example:
+
+| Airbyte Union Type                                                              | Avro Type                       | Parquet Type                                                                                  |
+|---------------------------------------------------------------------------------|---------------------------------|-----------------------------------------------------------------------------------------------|
+| `{"oneOf": [ {"type": "integer"}, {"type": "integer"} ] }`                      | `["null", "integer"]`           | `Optional Int64`                                                                              |
+| `{"oneOf": [ {"type": "integer"}, {"type": "string"}, {"type": "integer" } ] }` | `["null", "integer", "string"]` | `Optional Record { type: Optional String, integer: Optional Int64, string: Optional String }` |
+
+Unions of objects will continue to be merged into a single object with a union schema. If this is not possible due to field type conflicts, an exception will be thrown and the sync will fail.
+
+| Object 1 Properties           | Object 2 Properties             | Merged Object                                 |
+|-------------------------------|---------------------------------|-----------------------------------------------|
+| `{id: integer}`               | `{name: string}`                | `{id: integer, name: string}`                 |
+| `{id: integer, name: string}` | `{id: integer, birthday: date}` | `{id: integer, name: string, birthday: date}` |
+| `{id: integer}`               | `{id: string}`                  | [SYNC FAILED]                                 |
```

Comment:
I missed this in the pr review - would it be hard to generate `{id: [integer, string]}` for this case?

Reply:
Not hard, and an easy improvement. I *think* it's how the old code behaved, and I'd rather not add any more work at this point unless it's actually a regression. I'll double-check.

Reply:
Ugh, I missed that this is what the old code actually did. Added.